dask/blockwise.py

```diff
@@ -1098,8 +1103,8 @@ def make_blockwise_graph(
 
     # Apply Culling.
     # Only need to construct the specified set of output blocks
-    output_blocks = output_blocks or itertools.product(
-        *[range(dims[i]) for i in out_indices]
+    output_blocks = output_blocks or list(
+        itertools.product(*[range(dims[i]) for i in out_indices])
```

Comment:
We need `output_blocks` to be in memory (to check for dependencies) after the `for out_coords in output_blocks:` loop  below. The output of `itertools.product` is technically a generator, so this doesn't work unless we convert it to a list. This change does not mean that we will consume more memory, because the `itertools.product` ("generator") code was iterating through an in-memory list anyway.